[All] Remove use of compatibility layer

fredroy opened the pull request:
⚠! Uber PR incoming! ⚠
(Will basically conflict every time a branch is merged 🤪) 

This PR generalizes the use of the Sofa.Type module (with things formerly in SofaHelper, SofaDefaulttype and SofaCore).
E.g:
 - helper::fixed_array becomes type::fixed_array
 - defaulttype::Vec becomes type::Vec
 - defaulttype::Quat and helper::Quater becomes type::Quat (and typedef *Quaternion* is removed)
 - helper::types::RGBAColor becomes type::RGBAColor
 - and many more
More info in #1521 

These renames was handled in Sofa.Compat to be the less problematic possible while converting.
Now for the release of v21.06, it is logical to use the correct namespaces/headers.

Big Warning though: the Sofa.Compat is still here for the external plugins, but as the headers/file inclusions changes, it is highly probable that they will break at compilation: a lot of people does not "include what they use" and what relying on the previous inclusions.

Later TODO: add a cmake option to make the use of Sofa.Compat optional.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

fredroy committed:
re-enable warnings when using compat headers

fredroy committed:
sofaframework compiles

fredroy committed:
sofabase compiles

fredroy committed:
sofagl compiles

fredroy committed:
modules compiles

fredroy committed:
gui compiles

fredroy committed:
tests compiles

fredroy committed:
image

fredroy committed:
plugins

fredroy committed:
linux, plugins...

fredroy committed:
physicsapi, cuda

fredroy committed:
remove more uses

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
compiles without sofa.compat

fredroy committed:
compiles without sofa.compat with plugins

fredroy commented:
[ci-build]

fredroy committed:
Fix headlessrecorder

fredroy committed:
fix geomagic

fredroy committed:
fix sofaassimp

fredroy commented:
[ci-build]

fredroy commented:
[ci-build]

fredroy commented:
[ci-build][with-all-tests][force-full-build]

fredroy committed:
fix imagetoolbox

fredroy committed:
Fix compilation crash with masses

fredroy commented:
[ci-build][with-all-tests][force-full-build]

alxbilger commented on SofaKernel/modules/Sofa.GL/src/sofa/gl/BasicShapes.h:
```diff
@@ -175,7 +175,7 @@ void drawTorus(const float* coordinateMatrix, const float& bodyRad=0.0,  const f
 template <typename V>
 void drawEmptyParallelepiped(const V& vert1, const V& vert2, const V& vert3, const V& vert4, const V& vecFromFaceToOppositeFace, const float& rad=1.0, const int precision=8,
-                             const V& color=sofa::helper::fixed_array<int, 3>(255,0,0))
+                             const V& color=sofa::type::fixed_array<int, 3>(255,0,0))
```
```suggestion
                             const V& color = sofa::type::RGBAColor::red()
```

alxbilger commented on SofaKernel/modules/Sofa.Type/Sofa.Type_test/Quater_test.cpp:
```diff
@@ -330,8 +332,8 @@ TEST(QuaterTest, QuaterdOperatorDivRealSelf)
-//    Quater<double> quat, quatMult;
-//    sofa::defaulttype::Vec<3, double> vec(1,2,3);
+//    Quat<double> quat, quatMult;
+//    sofa::type::Vec<3, double> vec(1,2,3);
```
Should we keep commented code?

damienmarchal replied:
No never.

alxbilger commented on SofaKernel/modules/Sofa.Type/Sofa.Type_test/Quater_test.cpp:
```diff
@@ -340,8 +342,8 @@ TEST(QuaterTest, QuaterdOperatorDivRealSelf)
-//    Quater<double> quat, quatMult;
-//    sofa::defaulttype::Vec<3, double> vec(1, 2, 3);
+//    Quat<double> quat, quatMult;
+//    sofa::type::Vec<3, double> vec(1, 2, 3);
```
Should we keep this commented code?

fredroy replied:
obviously not but I wont clean the files 😴 in this PR

alxbilger commented on SofaKernel/modules/Sofa.Type/Sofa.Type_test/Quater_test.cpp:
```diff
@@ -517,10 +519,10 @@ TEST(QuaterTest, QuaterdCreateQuaterFromFrame)
 //{
 //    //axang2quat([1 2 5 pi / 12])
-//    sofa::defaulttype::Vec3d axis(1, 2, 5);
+//    sofa::type::Vec3d axis(1, 2, 5);
```
Should we keep commented code?

alxbilger commented on SofaKernel/modules/Sofa.Type/Sofa.Type_test/Quater_test.cpp:
```diff
@@ -544,10 +546,10 @@ TEST(QuaterTest, QuaterdCreateQuaterFromEuler)
 //{
 //    //axang2quat([1 2 5 pi / 12])
-//    sofa::defaulttype::Vec3d axis(1, 2, 5);
+//    sofa::type::Vec3d axis(1, 2, 5);
```
Should we keep commented code?

alxbilger commented on SofaKernel/modules/Sofa.Type/Sofa.Type_test/Quater_test.cpp:
```diff
@@ -574,8 +576,8 @@ TEST(QuaterTest, QuaterdQuatDiff)
 //{
-//    Quater<double> quat1, quat2;
-//    sofa::defaulttype::Vec3d displacement;
+//    Quat<double> quat1, quat2;
```
Should we keep commented code?

alxbilger commented on SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/IntrUtility3.cpp:
```diff
@@ -25,6 +25,7 @@
 {
 
+using namespace sofa::type;
 using namespace sofa::defaulttype;
```
Is this one still useful?

alxbilger commented on SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/SphereModel.cpp:
```diff
@@ -26,6 +26,7 @@
 {
 
+using namespace sofa::type;
 using namespace sofa::defaulttype;
```
Is it still useful?

alxbilger commented on SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/SphereModel.inl:
```diff
@@ -120,6 +120,7 @@ void SphereCollisionModel<DataTypes>::draw(const core::visual::VisualParams* vpa
         return ;
 
+    using namespace sofa::type;
     using namespace sofa::defaulttype;
```
Is it still useful?

alxbilger commented on SofaKernel/modules/SofaBaseLinearSolver/src/SofaBaseLinearSolver/matrix_bloc_traits.h:
```diff
@@ -120,39 +120,39 @@ class matrix_bloc_traits < defaulttype::Mat<L,C,real>, IndexType>
+//template<> inline const char* matrix_bloc_traits<type::Mat<9,9,float >, Size >::Name() { return "9f"; }
+//template<> inline const char* matrix_bloc_traits<type::Mat<9,9,double>, Size >::Name() { return "9d"; }
+//template<> inline const char* matrix_bloc_traits<type::Mat<12,12,float >, Size >::Name() { return "12f"; }
+//template<> inline const char* matrix_bloc_traits<type::Mat<12,12,double>, Size >::Name() { return "12d"; }
```
What do we do with all this commented code?

alxbilger commented on SofaKernel/modules/SofaBaseMechanics/src/SofaBaseMechanics/DiagonalMass.cpp:
```diff
@@ -33,6 +33,7 @@ namespace sofa::component::mass
 using sofa::core::objectmodel::ComponentState ;
 
+using namespace sofa::type;
 using namespace sofa::defaulttype;
```
Is `using namespace sofa::defaulttype;` still useful?

fredroy replied:
Actually in that particular file yes (look at the explicite instanciations with RigidTypes, etc) but  there are a lot of files where it wont be useful for sure (but I did not really want to check for the bazillon number of  files)

alxbilger commented on SofaKernel/modules/SofaBaseMechanics/src/SofaBaseMechanics/UniformMass.cpp:
```diff
@@ -34,6 +34,7 @@ using std::string ;
 using sofa::component::mass::Vec3d ;
 using sofa::helper::system::DataRepository ;
+using namespace sofa::type;
 using namespace sofa::defaulttype;
```
Is `using namespace sofa::defaulttype;` still useful?

alxbilger commented on SofaKernel/modules/SofaBaseTopology/SofaBaseTopology_test/TetrahedronNumericalIntegration_test.cpp:
```diff
@@ -36,6 +36,7 @@ using sofa::testing::NumericTest;
 
 using namespace component;
+using namespace type;
 using namespace defaulttype;
```
Is `using namespace sofa::defaulttype;` still useful?

alxbilger commented on SofaKernel/modules/SofaBaseTopology/SofaBaseTopology_test/TriangleNumericalIntegration_test.cpp:
```diff
@@ -37,6 +37,7 @@ using sofa::testing::NumericTest;
 
 using namespace component;
+using namespace type;
 using namespace defaulttype;
```
Is `using namespace sofa::defaulttype;` still useful?

fredroy committed:
Update SofaKernel/modules/Sofa.GL/src/sofa/gl/BasicShapes.h

Co-authored-by: Alex Bilger <alxbilger@users.noreply.github.com>

fredroy committed:
Merge remote-tracking branch 'upstream/master' into remove_compat_use

fredroy committed:
Merge remote-tracking branch 'upstream/master' into remove_compat_use

fredroy committed:
restore compat in cmake

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
fix compilation

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
Merge branch 'master' into remove_compat_use

fredroy committed:
fix sparseldl

guparan committed:
[All] FIX deprecated includes "sofa/helper/ArgumentParser.h"

# Conflicts:
#	applications/projects/SofaFlowVR/Main.cpp
#	applications/projects/runSofa/Main.cpp
#	applications/tutorials/oneParticle/oneParticle.cpp

guparan committed:
[All] FIX deprecated includes "sofa/helper/ArgumentParser.h" (2)
